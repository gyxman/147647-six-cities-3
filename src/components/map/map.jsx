import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import * as leaflet from "leaflet";

export class Map extends PureComponent {
  constructor(props) {
    super(props);

    this.mapRef = React.createRef();
  }

  componentDidMount() {
    const {offers} = this.props;

    const icon = leaflet.icon({
      iconUrl: `img/pin.svg`,
      iconSize: [30, 30]
    });

    const city = [52.38333, 4.9];

    const zoom = 12;
    const map = leaflet.map(this.mapRef.current, {
      center: city,
      zoom,
      zoomControl: false,
      marker: true
    });
    map.setView(city, zoom);

    leaflet
      .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
      .addTo(map);

    offers.forEach((offer) => {
      leaflet.marker(offer, {icon}).addTo(map);
    });
  }

  render() {
    const {className} = this.props;

    return (<section className={className + ` map`} id="map" ref={this.mapRef} />);
  }
}

Map.propTypes = {
  className: PropTypes.string.isRequired,
  offers: PropTypes.arrayOf(
      PropTypes.arrayOf(
          PropTypes.number.isRequired
      ).isRequired
  ).isRequired,
};
