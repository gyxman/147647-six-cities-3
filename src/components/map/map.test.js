import React from "react";
import renderer from "react-test-renderer";
import {Map} from "./map.jsx";

const mock = [
  [52.3809553943508, 4.939309666406198]
];

it(`Если приложение загрузилось, то компонент Map отрисовался`, () => {
  const tree = renderer
    .create(<Map
      offers={mock}
      className={`cities__map`}
    />, {
      createNodeMock: () => document.createElement(`div`)
    }).toJSON();

  expect(tree).toMatchSnapshot();
});
