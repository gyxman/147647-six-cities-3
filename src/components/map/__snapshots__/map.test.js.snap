// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Если приложение загрузилось, то компонент Map отрисовался 1`] = `
<section
  className="cities__map map"
  id="map"
/>
`;
